learning/step_by_step/scripting.rst

```diff
@@ -91,129 +101,126 @@ look like the image below. You can set the text in the Inspector pane.
-property:
+Once this is done, the script will be created and added to the node. You can
+see this both as an extra icon in the node as well as in the script property:
 
 .. image:: /img/script_added.png
 
-To edit the script, select either of the highlighted buttons. 
-This will bring you to the script editor where an existing template will
-be included by default:
+To edit the script, select either of the highlighted buttons. This will bring
+you to the script editor where an existing template will be included by default:
 
 .. image:: /img/script_template.png
 
 There is not much in there. The "_ready()" function is called when the
-node (and all its children) entered the active scene. (Remember, it's
-not a constructor, the constructor is "_init()").
+node (and all its children) enter the active scene. (Note: "_ready()" is not
+the script's constructor; the constructor is "_init()").
 
 The role of the script
 ~~~~~~~~~~~~~~~~~~~~~~
 
-A script adds behavior to a node. It is used to control the
-node functions as well as other nodes (children, parent, siblings, etc).
-The local scope of the script is the node (just like in regular
-inheritance) and the virtual functions of the node are captured by the
-script.
+A script adds behavior to a node. It is used to control how the node functions
+as well as how it interacts with other nodes (children, parent, siblings,
+etc.). The local scope of the script is the node. In other words, the script
+inherits the functions provided by that node.
 
 .. image:: /img/brainslug.jpg
 
 Handling a signal
 ~~~~~~~~~~~~~~~~~
 
-Signals are used mostly in GUI nodes (although other nodes have them
-too). Signals are "emitted" when some specific kind of action happens,
-and can be connected to any function of any script instance. In this
-step, the "pressed" signal from the button will be connected to a custom
-function.
+Signals are "emitted" when some specific kind of action happens, and they can be
+connected to any function of any script instance. Signals are used mostly in
+GUI nodes (although other nodes have them too, and you can even define custom
+signals in your own scripts).
 
-An interface for connecting signals to your scripts exists in the editor. 
-You can access this by selecting the node in the scene tree and then
-selecting the "Node" tab. Make sure that you have "Signals" selected.
+In this step, we'll connect the "pressed" signal to a custom function.
+
+The editor provides an interface for connecting signals to your scripts. You
+can access this by selecting the node in the scene tree and then selecting the
+"Node" tab. Next, make sure that you have "Signals" selected.
 
 .. image:: /img/signals.png
 
-In any case, at this point it is clear that we are interested in
-the "pressed" signal. Instead of using the visual interface, we will opt
-to code the connection.
+At this point, you could use the visual interface to hook up the "pressed"
```

Comment:
Sorry I appear to have lost old comments by doing a force push. Earlier here, you said:

"Order of this two sentences may be swapped, since first one is extra information for future tutorials."

However, the reason I used this order was the transition from the previous paragraph and image (which talk about "signals" in the visual UI" to the next paragraph (where we have the user start writing code). Therefore, I feel like changing the order of these sentences doesn't quite work. Did I miss something?